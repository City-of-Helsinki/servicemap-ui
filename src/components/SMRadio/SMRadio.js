import React, { useState } from 'react';
import PropTypes from 'prop-types';
import {
  FormControl, FormLabel, RadioGroup, FormControlLabel, Radio,
} from '@material-ui/core';

const SMRadio = ({
  controls, initialValue, name, label, onChange, ...rest
}) => {
  const [value, setValue] = useState(initialValue);

  return (
    <FormControl component="fieldset">
      {
        label
        && (
          <FormLabel component="legend">{label}</FormLabel>
        )
      }
      <RadioGroup
        name={name}
        value={value}
        onChange={(event, value) => {
          setValue(value);
          if (onChange) {
            onChange(event, value);
          }
        }}
        {...rest}
      >
        {
          controls.map(control => (
            <FormControlLabel
              control={<Radio />}
              disabled={!!control.disabled}
              key={control.value}
              value={control.value}
              label={control.label}
            />

          ))
        }
      </RadioGroup>
    </FormControl>
  );
};

SMRadio.propTypes = {
  'aria-label': PropTypes.string.isRequired,
  controls: PropTypes.arrayOf(PropTypes.shape({
    disabled: PropTypes.bool,
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
    label: PropTypes.string.isRequired,
  })).isRequired,
  initialValue: PropTypes.string,
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

SMRadio.defaultProps = {
  initialValue: null,
  label: null,
};

export default SMRadio;
