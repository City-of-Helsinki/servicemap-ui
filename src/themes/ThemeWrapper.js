import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { ThemeProvider } from '@mui/material/styles';
import themes from '.';

const mapStateToProps = (state) => {
  const { user } = state;
  return {
    theme: user.theme,
  };
};

// Component to handle theme changes
const ThemeHandler = ({ children, theme }) => (
// Get correct theme setting from store
  <ThemeProvider theme={theme === 'dark' ? themes.SMThemeDark : themes.SMTheme}>
    {children}
  </ThemeProvider>
);
ThemeHandler.propTypes = {
  children: PropTypes.node,
  theme: PropTypes.string.isRequired,
};
ThemeHandler.defaultProps = {
  children: null,
};

const ThemeWrapper = connect(
  mapStateToProps,
)(ThemeHandler);

export default ThemeWrapper;
