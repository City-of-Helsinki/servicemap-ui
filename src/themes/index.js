import { createMuiTheme } from '@material-ui/core';

const focusIndicator = {
  outline: '2px solid transparent',
  boxShadow: '0 0 0 2px rgb(255, 255, 255), 0 0 0 6px rgb(71, 131, 235), 0 0 0 8px rgb(255, 255, 255)',
};

const focusIndicatorDark = {
  outline: '2px solid transparent',
  boxShadow: '0 0 0 2px rgb(255, 255, 255), 0 0 0 6px rgb(0, 0, 0), 0 0 0 8px rgb(255, 255, 255)',
};

const props = {
  // Globally disable all default mui focus effects
  MuiButtonBase: {
    disableRipple: true,
  },
  MuiButton: {
    disableRipple: true,
    disableFocusRipple: true,
  },
  MuiTypography: {
    variant: 'body2',
  },
};

const overrides = theme => ({
  MuiButtonBase: {
    root: {
      // Default keyboard focus indicator for buttons
      '&$focusVisible': theme === 'dark' ? focusIndicatorDark : focusIndicator,
    },
  },
  MuiInputBase: {
    // Default keyboard focus indicator for input fields
    root: { '&$focused': theme === 'dark' ? focusIndicatorDark : focusIndicator },
  },
  MuiListItemText: {
    root: {
      marginTop: 0,
      marginBottom: 0,
    },
  },
  MuiSvgIcon: {
    root: {
      fontSize: 24,
    },
  },
  PrivateSwitchBase: {
    root: {
      padding: 12,
    },
  },
});

const breakpoints = {
  values: {
    xs: 0,
    sm: 700,
    md: 900,
    lg: 1280,
    xl: 1920,
  },
};

const spacing = 8;

const custom = {
  body2light: {
    color: 'rgba(0,0,0,0.7)',
  },
};

const zIndex = {
  behind: -1,
  forward: 50,
  sticky: 51,
  infront: 900,
};

const typography = {
  useNextVariants: true,
  fontSize: 16,
  // Use the system font instead of the default Roboto font.
  fontFamily: [
    'Lato',
  ].join(','),
  body1: {
    fontSize: '1.043rem',
    letterSpacing: '0.03125rem', // 0.5px
    lineHeight: '1.75rem',
  },
  body2: {
    fontSize: '0.913rem',
    letterSpacing: '0.015625rem', // 0.25px
    lineHeight: '1.25rem',
  },
  subtitle1: {
    fontSize: '1.03rem',
    fontWeight: 'bold',
    letterSpacing: '0.00937rem', // 0.15px
    lineHeight: '1.5rem',
  },
  caption: {
    fontSize: '0.7725rem', // 12.36px
    fontWeight: 'bold',
    letterSpacing: '0.025rem', // 0.4px
    lineHeight: '1rem',
    color: 'rgba(0,0,0,0.6)',
  },
  h1: {
    fontSize: '6.259rem',
    letterSpacing: '-0.09375em', // -1.5px
  },
  h2: {
    fontSize: '3.862rem',
    fontWeight: 'bold',
    letterSpacing: '-0.03125rem', // -0.5px
  },
  h3: {
    fontSize: '3.089rem',
    fontWeight: 'bold',
    letterSpacing: 0,
  },
  h4: {
    fontSize: '2.214rem',
    letterSpacing: '0.015625rem', // 0.25px
  },
  h5: {
    fontSize: '1.564rem',
    letterSpacing: 0,
  },
  h6: {
    fontSize: '1.288rem',
    fontWeight: 'bold',
    letterSpacing: '0.015625rem', // 0.25px
    lineHeight: '1.5rem',
  },
  button: {
    fontSize: '0.90125rem', // 14.42px
    fontWeight: 'bold',
    letterSpacing: '0.078125rem', // 1.25px
    lineHeight: '1rem', // 16px
    color: '#ffffff',
  },
  overline: {
    fontSize: '0.7725rem', // 12.36px
    fontWeight: 'bold',
    letterSpacing: '0.125rem', // 2px
    lineHeight: '1rem',
  },
};

// Color palette for normal theme
export const paletteDefault = {
  primary: {
    main: 'rgb(25, 100, 230)',
    highContrast: '#fff',
  },
  secondary: {
    main: 'rgb(43, 47, 57)',
    hover: '#1d39ad',
  },
  background: {
    main: 'linear-gradient(340.58deg, rgb(11, 123, 237) 0%, rgb(20, 108, 232) 67.04%, rgb(25, 100, 230) 100%)',
    plain: 'rgb(20, 108, 232)',
    front: 'linear-gradient(340.58deg, rgb(11, 123, 237) 0%, rgb(20, 108, 232) 67.04%, rgb(25, 100, 230) 100%)',
  },
  white: {
    light: '#f2f2f2',
    main: '#FFFFFF',
    dark: '#949494',
    contrastText: '#000',
  },
  detail: {
    main: 'rgb(25, 100, 230)',
    alpha: 'rgba(25,100,230,0.5)',
  },
  disabled: {
    main: 'rgb(239, 239, 239)',
    strong: '#898989',
  },
  focusBorder: {
    main: 'rgb(71, 131, 235)',
  },
  warning: {
    main: '#AF0021',
  },
};

// Color palette for dark theme
export const paletteDark = {
  primary: {
    main: '#353638',
    highContrast: '#fff',
  },
  secondary: {
    main: 'rgb(43, 47, 57)',
    hover: '#5b5d61',
  },
  background: {
    main: '#4A4A4C',
    plain: '#4A4A4C',
    front: 'linear-gradient(326.21deg, rgba(0, 0, 0, 0.79) 0%, rgba(71, 71, 71, 0.79) 100%)',
  },
  white: {
    light: '#f2f2f2',
    main: '#FFFFFF',
    dark: '#949494',
    contrastText: '#000',
  },
  detail: {
    main: '#585B63',
    alpha: 'rgba(88,91,99,0.5)',
  },
  disabled: {
    main: 'rgb(239, 239, 239)',
    strong: '#898989',
  },
  warning: {
    main: '#AF0021',
  },
  focusBorder: {
    main: '#000',
  },
};

// Themes
const SMTheme = createMuiTheme({
  props,
  overrides: overrides('default'),
  breakpoints,
  typography,
  spacing,
  custom,
  palette: paletteDefault,
  zIndex,
});

const SMThemeDark = createMuiTheme({
  props,
  overrides: overrides('dark'),
  breakpoints,
  typography,
  spacing,
  custom,
  palette: paletteDark,
  zIndex,
});


export default { SMTheme, SMThemeDark };
