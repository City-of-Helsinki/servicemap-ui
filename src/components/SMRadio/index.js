import SMRadio from './SMRadio';

export default SMRadio;
