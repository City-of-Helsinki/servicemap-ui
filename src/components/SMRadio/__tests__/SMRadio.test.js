// // Link.react.test.js
import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import { MuiThemeProvider, RadioGroup, FormControlLabel } from '@material-ui/core';
import themes from '../../../../themes';
import SMRadio from '../index';

// Generic required props for SimpleListItem
const mockProps = {
  'aria-label': 'Aria label text',
  name: 'service',
  initialValue: 'none',
  controls: ['none', 'common', 'all'].map(service => ({
    value: service,
    label: `label ${service}`,
  })),
  onChange: () => {},
};

// eslint-disable-next-line react/prop-types
const Providers = ({ children }) => (
  <MuiThemeProvider theme={themes.SMTheme}>
    {children}
  </MuiThemeProvider>
);

describe('<SMRadio />', () => {
  let mount;

  beforeEach(() => {
    mount = createShallow({ wrappingComponent: Providers });
  });

  it('should work', () => {
    const component = mount(<SMRadio {...mockProps} />);
    expect(component).toMatchSnapshot();
  });

  it('simulates onChange event correctly', () => {
    const mockCallBack = jest.fn();
    const component = mount(<SMRadio {...mockProps} onChange={mockCallBack} />);

    component.find(RadioGroup).simulate('change', { value: 'common' });
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });

  it('does set RadioGroup correctly', () => {
    const component = mount(<SMRadio {...mockProps} />);

    const radioGroup = component.find(RadioGroup);
    expect(radioGroup.props().name).toEqual(mockProps.name);
    expect(radioGroup.props().value).toEqual(mockProps.initialValue);
    expect(radioGroup.props()['aria-label']).toEqual(mockProps['aria-label']);
  });

  it('does set FormControlLabels correctly', () => {
    const component = mount(<SMRadio {...mockProps} />);

    const label = component.find(FormControlLabel);
    expect(label.at(0).props().label).toEqual(mockProps.controls[0].label);
    expect(label.at(0).props().value).toEqual(mockProps.controls[0].value);
    expect(label.at(1).props().label).toEqual(mockProps.controls[1].label);
    expect(label.at(1).props().value).toEqual(mockProps.controls[1].value);
    expect(label.at(2).props().label).toEqual(mockProps.controls[2].label);
    expect(label.at(2).props().value).toEqual(mockProps.controls[2].value);
  });
});
