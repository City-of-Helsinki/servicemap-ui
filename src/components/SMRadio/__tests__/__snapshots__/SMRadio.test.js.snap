// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SMRadio /> should work 1`] = `
<WithStyles(FormControl)
  component="fieldset"
>
  <RadioGroup
    aria-label="Aria label text"
    name="service"
    onChange={[Function]}
    value="none"
  >
    <WithStyles(WithFormControlContext(FormControlLabel))
      control={<WithStyles(Radio) />}
      disabled={false}
      key="none"
      label="label none"
      value="none"
    />
    <WithStyles(WithFormControlContext(FormControlLabel))
      control={<WithStyles(Radio) />}
      disabled={false}
      key="common"
      label="label common"
      value="common"
    />
    <WithStyles(WithFormControlContext(FormControlLabel))
      control={<WithStyles(Radio) />}
      disabled={false}
      key="all"
      label="label all"
      value="all"
    />
  </RadioGroup>
</WithStyles(FormControl)>
`;
